### ISSUE TITLE
POST on nested data throws 404

### ISSUE THREAD

**Issue by andneuma** [typicode/json-server#1213]
Hello everybody,

we're trying get json-server accepting a nested POST request to `/findingsets/` where `findings` are nested to. Unfortunately when doing `POST /findingsets/1/findings/` it throws `404` where we would expect that json-server adds our payload to the nested array. This is how our data structure looks like

```
{
  "findingsets": [
    {
      "id": 1,
      "name": "First findingSet",
      "findings": [ 
        { 
          "id": 888213,
          "name": "First finding (Backend)",
          "latitude": 2,
          "longitude": 6,
          "description": "Ich bin ein erster Platz"
        } 
      ] 
    },
    { 
      "id": 2,
      "name": "Second findingSet",
      "findings": [ 
        { 
          "id": 852033,
          "name": "First finding (Backend)",
          "latitude": 18,
          "longitude": 33,
          "description": "Ich bin ein zweiter Platz"
        },
        { 
          "id": 101543,
          "name": "Second finding (Backend)",
          "latitude": 26,
          "longitude": 30,
          "description": "Ich bin ein dritter Platz"
        },
        { 
          "id": 453135,
          "name": "Third finding (Backend)",
          "latitude": 19,
          "longitude": 30,
          "description": "Ich bin ein vierter Platz"
        } 
      ] 
    } 
}
```

Any idea what we're doing wrong here?

**Comment by crevulus** [comment]
Following the this thread because I have the same issue. I can GET from `/profile/1` just fine but trying to send a POST req to the same endpoint gives me a 404. POST req to the `/profile` endpoint without the id also works fine.

**Comment by shreym03** [comment]
Same problem with `POST /tasks/1/` and `PUT /tasks/` as well. It gives me a 404 Error as well in my Angular App. It doesn't work with Postman as well. Not an issue with `DELETE /tasks/1/` and `GET /tasks/` though

**Comment by crevulus** [comment]
@shreym03 you gotta use PATCH instead of POST if you want to update /tasks/1. Only use POST for adding to /tasks. Not very intuitive, but I'm sure they have their reasons for it.

**Comment by dimitri-bourreau** [comment]
Did anyone found a solution to this problem? 🙏

**Comment by crevulus** [comment]
@dimitri-bourreau see my comment above. This server doesn't work like a normal server, so to update existing records you need to send a `PATCH`.

**Comment by dimitri-bourreau** [comment]
@crevulus thanks, I've seen your comment and it doesn't help me.

I can not use `PATCH`. json-server is used in my case as some sort of a mock of a real server, for testing.

The very specific 404 on a `POST` on a "complex" path (`/this/is/complex`) is my issue.

Would you agree to say that json-server is not meant for this particular use? This is my fear.

**Comment by petterw03** [comment]
Yes, I've also found that the odd behavior of json-server means that it is useless for use as a mock server, unless your production server is also using json-server..

**Comment by dimitri-bourreau** [comment]
We agree, thank you for sharing your idea on this. 👍
